torchinductor/lowering.py

```diff
@@ -1313,32 +1313,62 @@ def fn(idx):
     )
 
 
+@register_lowering(aten.scatter_, type_promote=False)
+def scatter_(self, dim: int, index, src, **kwargs):
+    if kwargs:
+        assert kwargs["reduce"] in {"add", "multiply"}
+    return scatter_reduce_(self, dim, index, src, kwargs["reduce"] if kwargs else None)
+
+
 @register_lowering(aten.scatter_add_, type_promote=False)
 def scatter_add_(self, dim: int, index, src):
+    return scatter_reduce_(self, dim, index, src, "add")
+
+
+@register_lowering(aten.scatter_reduce_, type_promote=False)
+def scatter_reduce_(self, dim: int, index, src, reduction_type: str = None, **kwargs):
+    assert reduction_type is None or reduction_type in {"add", "sum"}
     assert isinstance(self, TensorBox)
     assert "int" in str(index.get_dtype())
     assert 0 <= dim < len(self.get_size())
-    src = to_dtype(src, self.get_dtype())
+
     self.realize()
     V.graph.realize_users_of(self.get_name())
 
+    include_self = kwargs["include_self"] if kwargs else True
     index_loader = index.make_loader()
+    if isinstance(src, TensorBox):
```

Comment:
better bring it inside `fn`, separating this logic makes it more confusing

Reply:
Fixed in the new commit.